src/firefly/js/visualize/ui/SelectAreaDropDownView.jsx

```diff
@@ -0,0 +1,127 @@
+
+import SELECTRECT_DD from 'html/images/icons-2014/28x28_Rect_DD.png';
+import SELECTCIRCLE_DD from 'html/images/icons-2014/28x28_Circle-ON_DD.png';
+import SELECT_NONE_DD from 'html/images/icons-2014/28x28_NoSelect_DD.png';
+import SELECT_RECT from 'html/images/icons-2014/marquee.png';
+import SELECT_CIRCLE from 'html/images/icons-2014/28x28_Circle.png';
+import SELECT_NONE from 'html/images/icons-2014/28x28_NoSelect.png';
+
+export const selectAreas = new Enum([ 'rect', 'circle', 'noselect']);
+export const selectAreaInfo = {
+    [selectAreas.noselect.key] : {
+        typeId: '',
+        iconId: SELECT_NONE,
+        iconIdSelect: SELECT_NONE_DD,
+        label: 'None',
+        tip: 'turn off area selection'
+    },
+    [selectAreas.rect.key] : {
+        typeId: SelectArea.TYPE_ID,
+        iconId: SELECT_RECT,
+        iconIdSelect: SELECTRECT_DD,
+        label: 'Rectangular Selection',
+        tip: 'select rectangular area'
+    },
+    [selectAreas.circle.key] : {
+        typeId: SelectArea.TYPE_ID,
+        iconId:  SELECT_CIRCLE,
+        iconIdSelect: SELECTCIRCLE_DD,
+        label: 'Elliptical Selection',
+        tip: 'select elliptical area',
+        params: {selectedShape: selectAreas.circle.key, handleColor: 'white'}
+    }
+};
+
+function updateSelect(pv, ddCB, value, preValue, allPlots=true) {
+    const detatchPreSelectArea = (areaToDetach) => {
+        const dl = getDrawLayerByType(getDlAry(), selectAreaInfo[areaToDetach].typeId);
+
+        if (dl) {
+            if (isDrawLayerAttached(dl, pv.plotId)) {
+                dispatchDetachLayerFromPlot(selectAreaInfo[areaToDetach].typeId, pv.plotId, allPlots, dl.destroyWhenAllDetached);
+            }
+        }
+    };
+
+    return ()=> {
+        if (ddCB && Object.keys(selectAreaInfo).includes(value)) {
+            ddCB(value);
+        }
+
+        if (!pv) return;
+
+
+        if (value !== selectAreas.noselect.key && value !== preValue) {
+            if (preValue !== selectAreas.noselect.key) {
+                detatchPreSelectArea(preValue);
+            }
+            let dl = getDrawLayerByType(getDlAry(), selectAreaInfo[value].typeId);
+
+            if (!dl) {
+                dl = dispatchCreateDrawLayer(selectAreaInfo[value].typeId, selectAreaInfo[value].params);
+            }
+
+            if (!isDrawLayerAttached(dl, pv.plotId)) {
+               dispatchAttachLayerToPlot(selectAreaInfo[value].typeId, pv.plotId, allPlots);
+            }
+        }
+        if (value === selectAreas.noselect.key && value !== preValue) {
+            detatchPreSelectArea(preValue);
+        }
+    };
+}
+
+export function SelectAreaDropDownView({plotView:pv, allPlots, dropDownCB, crtSelection}) {
+    var enabled = !!pv;
+    let sep = 1;
+
+    const ddCB = dropDownCB ? dropDownCB : null;
+
+    const selectAreaCommands = () => {
+        return [selectAreas.rect, selectAreas.circle, selectAreas.noselect].reduce((prev, s) => {
```

Comment:
why use reduce and not map?

Reply:
I used 'reduce' since two tags are pushed into the result array in each iteration, it would be nice if 'map' can make it.